From 22fd6840660e5154a7051647051deefea90bfe27 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Wed, 26 Feb 2014 12:43:02 +0300
Subject: Fix a crash when removing displays

When a screen is extended to the right and then changed to mirrored
this can cause xfdesktop to crash. This patch does a sanity check
to ensure there is a backdrop when the backdrop_changed_cb is
called.

diff --git a/src/xfce-desktop.c b/src/xfce-desktop.c
index 9be8504..53be20e 100644
--- a/src/xfce-desktop.c
+++ b/src/xfce-desktop.c
@@ -367,7 +367,10 @@ backdrop_changed_cb(XfceBackdrop *backdrop, gpointer user_data)
     TRACE("entering");
     
     g_return_if_fail(XFCE_IS_DESKTOP(desktop));
-    
+
+    if(!XFCE_IS_BACKDROP(backdrop))
+        return;
+
     if(desktop->priv->updates_frozen || !gtk_widget_get_realized(GTK_WIDGET(desktop)))
         return;
 
-- 
cgit v0.10.1

